compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -11250,20 +11250,29 @@ private STNode parseTemplateContentAsRegExp() {
         // Separate out the interpolated expressions to a queue. Then merge the string content using '${}'.
         // These '${}' are used to represent the interpolated locations. Regular expression parser will replace '${}'
         // with the actual interpolated expression, while building the regular expression tree.
+        this.tokenReader.startMode(ParserMode.REGEXP);
         ArrayDeque<STNode> expressions = new ArrayDeque<>();
         StringBuilder regExpStringBuilder = new StringBuilder();
         STToken nextToken = peek();
+        boolean visitedCharacterClassStartToken = nextToken.text().equals("[");
         while (!isEndOfBacktickContent(nextToken.kind)) {
             STNode contentItem = parseTemplateItem();
             if (contentItem.kind == SyntaxKind.TEMPLATE_STRING) {
                 regExpStringBuilder.append(((STToken) contentItem).text());
+            } else if (visitedCharacterClassStartToken) {
+                regExpStringBuilder.append(contentItem);
             } else {
                 regExpStringBuilder.append("${}");
                 expressions.add(contentItem);
             }
             nextToken = peek();
+            if (nextToken.text().equals("[")) {
```

Comment:
When `${}` used in a character class in a capturing group they don't seem to work correctly.
```ballerina
_ = re `[^a-b${b-a}]`; // works as expected, start char code is greater than end char code
_ = re `([^a-b${b-a}])`; // no error, parser seem to treat `${b-a}` as an interpolation 
```

Reply:
Handled it and added tests in b5a73621343f655e48bb2d475fa63b98f5e70ad7